src/main/java/servlets/CustomerLoginServlet.java

```diff
@@ -16,7 +16,12 @@
 import com.bittercode.service.UserService;
 import com.bittercode.service.impl.UserServiceImpl;
 
-public class CustomerLoginServlet extends HttpServlet {
+// Suppressing Warnings here due to the fact that the CF believes that the "user.getFirstName"
+// call results in a potentially null return however the "login" method that preceeds it will either
+// return a null user which then the access never get called or the implementation of the "user" method
+// guarantees that the first name is initialized or else user will be null.

```

Comment:
This suppression shouldn't be necessary, based on the argument you're giving here (if `User` is annotated correctly, which I suspect that it isn't).